app/main.py

```diff
@@ -1,36 +1,105 @@
+from dotenv import load_dotenv
 import io
-from typing import Union
+import os
+import logging
+from typing import Annotated
 
-from fastapi import FastAPI, Response
+from fastapi import FastAPI, Header, HTTPException, Depends
 from fastapi.responses import StreamingResponse
+from fastapi.middleware.cors import CORSMiddleware
+from fastapi.middleware.trustedhost import TrustedHostMiddleware
 from pydantic import BaseModel
+from functools import lru_cache
+from weasyprint import HTML, urls
 
-from weasyprint import HTML
+# Set up logging
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+load_dotenv()
+
+class Settings:
+    def __init__(self):
+        self.API_KEY = os.getenv("API_KEY")
+        self.ALLOWED_HOSTS = os.getenv("ALLOWED_HOSTS", "*").split(",")
+        self.CORS_ORIGINS = os.getenv("CORS_ORIGINS", "*").split(",")
+
+@lru_cache()
+def get_settings():
+    return Settings()
 
 class PrintPdfRequest(BaseModel):
     html: str
     filename: str = 'weasyprint'
 
-# https://fastapi.tiangolo.com/
-app = FastAPI()
+app = FastAPI(
+    title="WeasyPrint PDF API",
+    description="A REST API for generating PDFs from HTML using WeasyPrint",
+    version="1.0.0"
```

Comment:
Version should not be assigned here. Leave it empty or we should assign it when we release a new version.